modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/app/TessellationIOApp.scala

```diff
@@ -72,6 +73,8 @@ abstract class TessellationIOApp[A <: CliMethod](
     */
   val kryoRegistrar: Map[Class[_], KryoRegistrationId[KryoRegistrationIdRange]]
 
+  val networkStateAfterJoining: NodeState
```

Comment:
@marcinwadon we may wish to discuss this variable as I am not really a fan of having to pull it out to this level. It was done because previously we were relying on the assembly process to merge the various `application.conf` files for each app for us. Since the config parameters were merged, the config reader (line 94) had access to the networkState parameter. However, since I renamed the app configs to avoid collisions within the bundled `sdk` jar, this networkState parameter is no longer merged into an `application.conf` file and thus is not present at this layer. I couldn't think of any other way to get this information so I added this `val` and specify it within each app.

If you (or anyone else) can think of a way to avoid this I would be appreciative.